test/test_sink.py

```diff
@@ -158,21 +183,128 @@ def testImageCopySmallMultiband(file_type, tmp_path):
     metadata = sink.getMetadata()
     assert metadata.get('sizeX') == 512
     assert metadata.get('sizeY') == 1024
-    assert metadata.get('dtype') == 'uint16'
+    assert metadata.get('dtype') == 'uint8'
     assert metadata.get('levels') == 2
     assert metadata.get('bandCount') == 7
     assert len(metadata.get('frames')) == 6
 
-    # TODO: fix these failures; unexpected metadata when reading it back
-    sink.write(output_file)
-    if file_type == 'zarr':
-        output_file /= '.zattrs'
+    sink.write(output_file, resample=resample_method)
     written = large_image.open(output_file)
     new_metadata = written.getMetadata()
 
     assert new_metadata.get('sizeX') == 512
     assert new_metadata.get('sizeY') == 1024
-    assert new_metadata.get('dtype') == 'uint16'
+    assert new_metadata.get('dtype') == 'uint8'
     assert new_metadata.get('levels') == 2 or new_metadata.get('levels') == 3
     assert new_metadata.get('bandCount') == 7
     assert len(new_metadata.get('frames')) == 6
+
+    written_arrays = dict(written._zarr.arrays())
+    assert len(written_arrays) == written.levels
+    assert written_arrays.get('0') is not None
+    assert written_arrays.get('0').shape == (2, 3, 1024, 512, 7)
+    assert written_arrays.get('1') is not None
+    assert written_arrays.get('1').shape == (2, 3, 512, 256, 7)
+
+
+@pytest.mark.parametrize('resample_method', list(ResampleMethod))
+def testImageCopySmallDownsampling(resample_method, tmp_path):
+    output_file = tmp_path / f'test_{resample_method}.db'
+    sink = large_image_source_zarr.new()
+    source = large_image_source_test.TestTileSource(
+        fractal=True,
+        tileWidth=128,
+        tileHeight=128,
+        sizeX=512,
+        sizeY=1024,
```

Comment:
We have pretty good coverage, but aren't exercising the odd-sized-tiles code.  We could use odd dimensions to exercise it; making the sizes slightly smaller would make the tests take no more time than at present; there is no reason they need to be powers of 2.  Obviously the asserts would have to change, too.

Reply:
Ok, I changed the dimension sizes by 1 in [2e10ec4](https://github.com/girder/large_image/pull/1490/commits/2e10ec443da02cb4f454a64867090f71034826cd)